research/workflow-engines/argo-workflows.md

```diff
@@ -0,0 +1,63 @@
+---
+title: Argo Workflows
+authors: lbarcziova
+---
+
+- docs: https://argoproj.github.io/argo-workflows/
+
+  > Argo Workflows is an open source container-native workflow engine for orchestrating parallel jobs on Kubernetes. Argo Workflows is implemented as a Kubernetes CRD (Custom Resource Definition).
+  >
+  > - Define workflows where each step in the workflow is a container.
+  > - Model multi-step workflows as a sequence of tasks or capture the dependencies between tasks using a directed acyclic graph (DAG).
+  > - Easily run compute intensive jobs for machine learning or data processing in a fraction of the time using Argo Workflows on Kubernetes.
+
+- example of multi-step workflow config:
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: steps-
+spec:
+  entrypoint: hello-hello-hello
```

Comment:
Could you please try to transform this example into our use cases so we can better imagine what this would look like?

(I really like how you've shown it on Airflow.)

Reply:
I did not see a benefit in showing this here as this is a static yaml configuration and doesn't fit the nature of dynamically creating tasks that we do

but I will check how some client library could be used for dynamically scheduling the tasks and extend this